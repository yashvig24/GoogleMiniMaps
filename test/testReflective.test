# Test MarvelPaths does not allow adding reflexive edge

# Parse through the data and build the graph
LoadGraph reflective_nodes_graph reflective.tsv

# List nodes and edges to see if the graph is built correctly
ListNodes reflective_nodes_graph
ListChildren reflective_nodes_graph New-York
ListChildren reflective_nodes_graph San-Fran

# Find the shortest path 
FindPath reflective_nodes_graph New-York San-Fran

# Find the shortest path other side
FindPath reflective_nodes_graph San-Fran New-York

# Find the shortest path to itself
FindPath reflective_nodes_graph San-Fran San-Fran