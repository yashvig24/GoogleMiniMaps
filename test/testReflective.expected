# Test MarvelPaths does not allow adding reflexive edge

# Parse through the data and build the graph
loaded graph reflective_nodes_graph

# List nodes and edges to see if the graph is built correctly
reflective_nodes_graph contains: New-York San-Fran
the children of New-York in reflective_nodes_graph are: San-Fran(1.0)
the children of San-Fran in reflective_nodes_graph are: New-York(1.0)

# Find the shortest path
path from New-York to San-Fran:
New-York to San-Fran with weight 1.000
total cost: 1.000

# Find the shortest path other side
path from San-Fran to New-York:
San-Fran to New-York with weight 1.000
total cost: 1.000

# Find the shortest path to itself
path from San-Fran to San-Fran:
total cost: 0.000